dix/req.c

```diff
@@ -1418,7 +1436,8 @@ static void dix_item_rc_update(struct m0_dix_req  *req,
 		case DIX_GET:
 			m0_cas_get_rep(creq, key_idx, &get_rep);
 			rc = get_rep.cge_rc;
-			if (rc == 0) {
+			if (rc == 0 && dix_item_version_cmp(ditem, &get_rep)) {
+				m0_buf_free(&ditem->dxi_val);
```

Comment:
There was a memory leak here?
And this patch will fix it, right?

Reply:
The rest of the patch looks fine.

Reply:
Right, previously get requests went to a single cas service. This PR changes the behavior to send to multiple replicas. `dix_item_rc_update` assumed exactly one reply which would be freed elsewhere, and would leak the values from additional replies. This version explicitly frees values when we get a more recent version (since the previous call to `m0_cas_rep_mlock` prevents these from being cleaned up during finalization.)

Reply:
Thanks for this explanation. I was not aware of this.